hw/bsp/dialog_da1469x-dk-pro/otp_tool.py

```diff
@@ -0,0 +1,538 @@
+        data += f_bytes
+        data += struct.pack('H', crc16(f_bytes, 0, trans_length))
+
+        try:
+            ser.write(data)
+        except serial.SerialException:
+            raise SystemExit("Failed to write cmd to %s" % uart)
+
+        data = ser.read(16)
+        if len(data) != 16:
+            raise SystemExit("Failed to receive response, exiting")
+
+        response = cmd_response._make(struct.unpack_from('IIII', data))
+        if response.status == 0:
+            data = ser.read(response.length)
+            if len(data) != response.length:
+                raise SystemExit("Failed to receive response, exiting")
+            f.write(data)
+
+        else:
+            SystemExit("Error in read response, exiting")
+            break
+
+        bytes_left -= trans_length
+        offset += trans_length
+
+    print("Successfully wrote flash")
+
+
+def send_otp_config_payload(uart, data):
+    try:
+        ser = serial.Serial(port=uart, baudrate=115200, timeout=1,
+                            bytesize=8, stopbits=serial.STOPBITS_ONE)
+    except serial.SerialException:
+        raise SystemExit("Failed to open serial port")
+
+    data_bytes = generate_payload(data)
+
+    # length is unused, so set to 0
+    cmd = cmd_append_value(0xaa55aa55, Cmd.OTP_APPEND_VALUE,
+                           len(data_bytes) + 2)
+    msg = struct.pack('III', *cmd)
+
+    msg += data_bytes
+    msg += struct.pack('H', crc16(data_bytes, 0, len(data_bytes)))
+
+    try:
+        ser.write(msg)
+    except serial.SerialException:
+        raise SystemExit("Failed to write to %s" % uart)
+
+    data = ser.read(16)
+    if len(data) != 16:
+        raise SystemExit("Failed to receive response, exiting")
+
+    response = cmd_response._make(struct.unpack_from('IIII', data))
+    if response.status == 0:
+        data = ser.read(response.length)
+        if len(data) != response.length:
+            raise SystemExit("Failed to receive data, exiting")
+
+
+@click.command(help='Append register value to OTP configuration script')
+@click.option('-a', '--address', type=str, required=True,
+              help='register address in hexadecimal')
+@click.option('-v', '--value', type=str, required=True,
+              help='register value in hexadecimal')
+@click.option('-u', '--uart', required=True, help='uart port')
+def otp_append_register(uart, address, value):
+
+    send_otp_config_payload(uart, [int(address, 16), int(value, 16)])
+
+
+@click.option('-t', '--trim', type=str, required=True,
+              multiple=True, help='trim value')
+@click.option('-i', '--index', type=int, required=True,
+              help='Trim value id')
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Append trim value to OTP configuration script')
+def otp_append_trim(uart, trim, index):
+
+    data = []
+    data.append(0x90000000 + (len(trim) << 8) + index)
+    for entry in trim:
+        data.append(int(entry, 16))
```

Comment:
Although this does not need to be efficient, one way to avoid the reallocation while expanding the list would be:

```
    data = [0 * (len(trim) + 1)]
    data[0] = 0x90000000 + (len(trim) << 8) + index
    for i, entry in enumerate(trim):
        data[i+1] = int(entry, 16)
```